includes/AnalysisSettings/StaticAnalysisSettings.js

```diff
@@ -0,0 +1,156 @@
+
+function StaticAnalysisType(type) {
+  const StaticAnalysisType_dict = {
+    undefined           : "GEOMETRICALLY_LINEAR",
+    "linear":             "GEOMETRICALLY_LINEAR",
+    "second order":       "SECOND_ORDER_P_DELTA",
+    "large deformations": "LARGE_DEFORMATIONS"
+  };
+
+  var SASType = StaticAnalysisType_dict[type];
+  if (SASType === undefined) {
+    SASType = "GEOMETRICALLY_LINEAR";
+    console.log("Wrong static analysis type input. Value was: " + type);
+    console.log("Correct values are: ('linear', 'second order', 'large deformations')");
+  };
+  console.log("Static analysis type: " + SASType);
+  return SASType;  
+}
+
+function NonlinearMethods(type, method) {
+
+  const nonlinearMethods_secondOrder_dict = {
+  "Picard"        : "PICARD",
+  "Postcritical"  : "NEWTON_RAPHSON_WITH_POSTCRITICAL_ANALYSIS",
+  "Newton"        : "NEWTON_RAPHSON"
+  };
+
+  const nonlinearMethods_largeDeformations_dict = {
+    "Picard"             : "PICARD",
+    "Combined"           : "NEWTON_RAPHSON_COMBINED_WITH_PICARD",
+    "Postcritical"       : "NEWTON_RAPHSON_WITH_POSTCRITICAL_ANALYSIS",
+    "Newton"             : "NEWTON_RAPHSON",
+    "Constant stiffness" : "NEWTON_RAPHSON_WITH_CONSTANT_STIFFNESS",
+    "Dynamic"            : "DYNAMIC_RELAXATION",
+  };
+
+  const nonlinearMehodsSwitcher = {
+    "Second-order (P-Δ)"   : nonlinearMethods_secondOrder_dict,
+    "Large deformations"   : nonlinearMethods_largeDeformations_dict, 
+  };
+
+  const method_dict = nonlinearMehodsSwitcher[type];
+  if (method_dict === undefined) {
+    console.log("It is not possible to set nonlinear analysis method for analysis type: " + type);
+    const nonlinear_method = undefined;
+  }
+  else {
+    const nonlinear_method = method_dict[method];
+    if (nonlinear_method === undefined) {
+      nonlinear_method = "NEWTON_RAPHSON";
+      console.log("Wrong nonlinear analysis method input. Value was: " + method);
+      console.log("Correct values are: ( " + Object.keys(method_dict) + ")");
+    };
+  };
+  return nonlinear_method;  
+};
+
+
+function SetEquationSolver(solverType) {
+
+  const EquationSolver_dict = {
+  "direct"    : "METHOD_OF_EQUATION_SYSTEM_DIRECT",
+  "iterative" : "METHOD_OF_EQUATION_SYSTEM_ITERATIVE"
+  };
+
+  var equationSolver = EquationSolver_dict[solverType];
+  if (equationSolver === undefined) {
+    console.log("Wrong equation solver input. Value was: " + solverType);
+    console.log("Correct values are: ( " + Object.keys(EquationSolver_dict) + ")");
+    equationSolver = "METHOD_OF_EQUATION_SYSTEM_DIRECT";
+  };
+  return equationSolver 
+};
+
+
+// Main function
+function StaticAnalysisSettings(no,
```

Comment:
Would it be possible to create separate prototype methods based on analysis type - Geometrically linear, Second Order and Large deformation. Then each function has its own argument which could be clear to user.

Reply:
New prototypes created